owncloudData/src/test/java/com/owncloud/android/data/server/datasources/implementation/OCRemoteServerInfoDataSourceTest.kt

```diff
@@ -69,6 +69,39 @@ class OCRemoteServerInfoDataSourceTest {
         every { clientManager.getClientForUnExistingAccount(any(), any()) } returns ocClientMocked
     }
 
+    @Test
+    fun getAuthenticationMethodFollowRedirections() {
+        val checkPathExistenceResultFollowRedirectionMocked: RemoteOperationResult<Boolean> =
+            createRemoteOperationResultMock(
+                data = true,
+                isSuccess = true,
+                redirectedLocation = OC_SERVER_INFO.baseUrl
+            )
+        val checkPathExistenceResultMocked: RemoteOperationResult<Boolean> =
+            createRemoteOperationResultMock(
+                data = true,
+                isSuccess = true,
+                resultCode = OK_SSL,
+                authenticationHeader = authHeadersBasic,
+                httpCode = HTTP_UNAUTHORIZED
+            )
+
+        every {
+            ocServerInfoService.checkPathExistence(redirectedLocation, false, ocClientMocked)
+        } returns checkPathExistenceResultFollowRedirectionMocked
+
+        every {
+            ocServerInfoService.checkPathExistence(OC_SERVER_INFO.baseUrl, false, ocClientMocked)
+        } returns checkPathExistenceResultMocked
+
+        val authenticationMethod = ocRemoteServerInfoDatasource.getAuthenticationMethod(redirectedLocation)
+
+        assertNotNull(authenticationMethod)
+        assertEquals(AuthenticationMethod.BASIC_HTTP_AUTH, authenticationMethod)
+
+        verify { ocServerInfoService.checkPathExistence(OC_SERVER_INFO.baseUrl, false, ocClientMocked) }
+    }
+
```

Comment:
Related to the previous comment. If previous one is required, then keep this test, if not, we can remove it 👍